api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -99,15 +99,22 @@
      * The configuration value is not guaranteed to be cached by the implementation, and may be expensive
      * to compute; therefore, if the returned value is intended to be frequently used, callers should consider storing
      * rather than recomputing it.
+     * <p>
+     * The result of this method is identical to the result of calling {@code getOptionalValue(propertyName, propertyType).get()}.
+     * In particular, If the given property name or the value element of this property does not exist, the {@link NoSuchElementException} is thrown.
+     * This method never returns {@code null}.
      *
-     * @param <T> The property type
-     * @param propertyName The configuration property name
-     * @param propertyType The type into which the resolved property value should get converted
-     * @return the resolved property value as an instance of the requested type
-     * @throws java.lang.IllegalArgumentException if the property cannot be converted to the specified type
-     * @throws java.util.NoSuchElementException if the property isn't present in the configuration
+     * @param <T>
+     *             The property type
+     * @param propertyName
+     *             The configuration property name
+     * @param propertyType
+     *             The type into which the resolved property value should get converted
+     * @return the resolved property value as an instance of the requested type (not {@code null})
+     * @throws IllegalArgumentException if the property cannot be converted to the specified type
+     * @throws NoSuchElementException if the property name or the value of the property does not exist
      */
-    <T> T getValue(String propertyName, Class<T> propertyType);
+    <T> T getValue(String propertyName, Class<T> propertyType) throws IllegalArgumentException, NoSuchElementException;
```

Comment:
I think we shouldn't declare the exceptions in the method signature. We were not doing it before and they are unchecked.

Reply:
ok, I agree with you. I think I inherited from the previous PR.